diff --git a/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldInstanceTest.php b/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldInstanceTest.php
index 561e162460..6696b0f572 100644
--- a/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldInstanceTest.php
+++ b/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldInstanceTest.php
@@ -130,6 +130,7 @@ public function testFieldInstances() {
     $this->assertEntity('node.test_content_type.field_long_text', 'Long text', 'text_with_summary', FALSE, FALSE);
     $this->assertEntity('node.test_content_type.field_term_reference', 'Term Reference', 'entity_reference', FALSE, FALSE);
     $this->assertEntity('node.test_content_type.field_text', 'Text', 'string', FALSE, FALSE);
+    $this->assertEntity('node.test_content_type.field_telephone', 'Telephone', 'telephone', FALSE, FALSE);
     $this->assertEntity('comment.comment_node_test_content_type.field_integer', 'Integer', 'integer', FALSE, TRUE);
     $this->assertEntity('user.user.field_file', 'File', 'file', FALSE, FALSE);
 
diff --git a/core/modules/migrate/tests/src/Kernel/Plugin/MigrationProvidersExistTest.php b/core/modules/migrate/tests/src/Kernel/Plugin/MigrationProvidersExistTest.php
index 194f044a98..892c542c14 100644
--- a/core/modules/migrate/tests/src/Kernel/Plugin/MigrationProvidersExistTest.php
+++ b/core/modules/migrate/tests/src/Kernel/Plugin/MigrationProvidersExistTest.php
@@ -100,6 +100,10 @@ public function testFieldProvidersExist() {
         'source_module' => 'phone',
         'destination_module' => 'telephone',
       ],
+      'telephone' => [
+        'source_module' => 'telephone',
+        'destination_module' => 'telephone',
+      ],
       'link' => [
         'source_module' => 'link',
         'destination_module' => 'link',
diff --git a/core/modules/migrate_drupal/tests/fixtures/drupal7.php b/core/modules/migrate_drupal/tests/fixtures/drupal7.php
index 8df6422c67..61977aaa8b 100644
--- a/core/modules/migrate_drupal/tests/fixtures/drupal7.php
+++ b/core/modules/migrate_drupal/tests/fixtures/drupal7.php
@@ -4294,6 +4294,21 @@
   'translatable' => '1',
   'deleted' => '0',
 ))
+->values(array(
+  'id' => '53',
+  'field_name' => 'field_telephone',
+  'type' => 'telephone',
+  'module' => 'telephone',
+  'active' => '1',
+  'storage_type' => 'field_sql_storage',
+  'storage_module' => 'field_sql_storage',
+  'storage_active' => '1',
+  'locked' => '0',
+  'data' => 'a:7:{s:12:"translatable";i:0;s:12:"entity_types";a:0:{}s:8:"settings";a:1:{s:23:"entity_translation_sync";b:0;}s:7:"storage";a:5:{s:4:"type";s:17:"field_sql_storage";s:8:"settings";a:0:{}s:6:"module";s:17:"field_sql_storage";s:6:"active";s:1:"1";s:7:"details";a:1:{s:3:"sql";a:2:{s:18:"FIELD_LOAD_CURRENT";a:1:{s:26:"field_data_field_telephone";a:1:{s:5:"value";s:21:"field_telephone_value";}}s:19:"FIELD_LOAD_REVISION";a:1:{s:30:"field_revision_field_telephone";a:1:{s:5:"value";s:21:"field_telephone_value";}}}}}s:12:"foreign keys";a:0:{}s:7:"indexes";a:0:{}s:2:"id";s:2:"53";}',
+  'cardinality' => '1',
+  'translatable' => '0',
+  'deleted' => '0',
+))
 ->execute();
 $connection->schema()->createTable('field_config_instance', array(
   'fields' => array(
@@ -5028,6 +5043,15 @@
   'data' => 'a:6:{s:5:"label";s:7:"Boolean";s:6:"widget";a:4:{s:4:"type";s:13:"options_onoff";s:6:"weight";s:2:"14";s:8:"settings";a:1:{s:13:"display_label";i:0;}s:6:"module";s:7:"options";}s:8:"settings";a:2:{s:18:"user_register_form";b:0;s:23:"entity_translation_sync";b:0;}s:7:"display";a:1:{s:7:"default";a:5:{s:5:"label";s:5:"above";s:4:"type";s:12:"list_default";s:8:"settings";a:0:{}s:6:"module";s:4:"list";s:6:"weight";i:12;}}s:8:"required";b:0;s:11:"description";s:0:"";}',
   'deleted' => '0',
 ))
+->values(array(
+  'id' => '82',
+  'field_id' => '53',
+  'field_name' => 'field_telephone',
+  'entity_type' => 'node',
+  'bundle' => 'test_content_type',
+  'data' => 'a:7:{s:5:"label";s:9:"Telephone";s:6:"widget";a:5:{s:6:"weight";s:2:"21";s:4:"type";s:17:"telephone_default";s:6:"module";s:9:"telephone";s:6:"active";i:1;s:8:"settings";a:1:{s:11:"placeholder";s:0:"";}}s:8:"settings";a:2:{s:18:"user_register_form";b:0;s:23:"entity_translation_sync";b:0;}s:7:"display";a:2:{s:7:"default";a:5:{s:5:"label";s:5:"above";s:4:"type";s:14:"telephone_link";s:6:"weight";s:2:"21";s:8:"settings";a:1:{s:5:"title";s:0:"";}s:6:"module";s:9:"telephone";}s:6:"teaser";a:5:{s:5:"label";s:5:"above";s:4:"type";s:10:"text_plain";s:6:"weight";s:1:"0";s:8:"settings";a:0:{}s:6:"module";s:4:"text";}}s:8:"required";i:0;s:11:"description";s:0:"";s:13:"default_value";N;}',
+  'deleted' => '0',
+))
 ->execute();
 $connection->schema()->createTable('field_data_body', array(
   'fields' => array(
@@ -7927,6 +7951,108 @@
   'field_tags_tid' => '17',
 ))
 ->execute();
+$connection->schema()->createTable('field_data_field_telephone', array(
+  'fields' => array(
+    'entity_type' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '128',
+      'default' => '',
+    ),
+    'bundle' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '128',
+      'default' => '',
+    ),
+    'deleted' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'tiny',
+      'default' => '0',
+    ),
+    'entity_id' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'revision_id' => array(
+      'type' => 'int',
+      'not null' => FALSE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'language' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '32',
+      'default' => '',
+    ),
+    'delta' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'field_telephone_value' => array(
+      'type' => 'varchar',
+      'not null' => FALSE,
+      'length' => '256',
+    ),
+  ),
+  'primary key' => array(
+    'entity_type',
+    'entity_id',
+    'deleted',
+    'delta',
+    'language',
+  ),
+  'indexes' => array(
+    'entity_type' => array(
+      'entity_type',
+    ),
+    'bundle' => array(
+      'bundle',
+    ),
+    'deleted' => array(
+      'deleted',
+    ),
+    'entity_id' => array(
+      'entity_id',
+    ),
+    'revision_id' => array(
+      'revision_id',
+    ),
+    'language' => array(
+      'language',
+    ),
+  ),
+  'mysql_character_set' => 'utf8',
+));
+
+$connection->insert('field_data_field_telephone')
+  ->fields(array(
+    'entity_type',
+    'bundle',
+    'deleted',
+    'entity_id',
+    'revision_id',
+    'language',
+    'delta',
+    'field_telephone_value',
+  ))
+  ->values(array(
+    'entity_type' => 'node',
+    'bundle' => 'test_content_type',
+    'deleted' => '0',
+    'entity_id' => '1',
+    'revision_id' => '1',
+    'language' => 'und',
+    'delta' => '0',
+    'field_telephone_value' => '99-99-99-99',
+  ))
+  ->execute();
 $connection->schema()->createTable('field_data_field_term_entityreference', array(
   'fields' => array(
     'entity_type' => array(
@@ -13127,6 +13253,109 @@
   'field_tags_tid' => '17',
 ))
 ->execute();
+$connection->schema()->createTable('field_revision_field_telephone', array(
+  'fields' => array(
+    'entity_type' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '128',
+      'default' => '',
+    ),
+    'bundle' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '128',
+      'default' => '',
+    ),
+    'deleted' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'tiny',
+      'default' => '0',
+    ),
+    'entity_id' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'revision_id' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'language' => array(
+      'type' => 'varchar',
+      'not null' => TRUE,
+      'length' => '32',
+      'default' => '',
+    ),
+    'delta' => array(
+      'type' => 'int',
+      'not null' => TRUE,
+      'size' => 'normal',
+      'unsigned' => TRUE,
+    ),
+    'field_telephone_value' => array(
+      'type' => 'varchar',
+      'not null' => FALSE,
+      'length' => '256',
+    ),
+  ),
+  'primary key' => array(
+    'entity_type',
+    'entity_id',
+    'revision_id',
+    'deleted',
+    'delta',
+    'language',
+  ),
+  'indexes' => array(
+    'entity_type' => array(
+      'entity_type',
+    ),
+    'bundle' => array(
+      'bundle',
+    ),
+    'deleted' => array(
+      'deleted',
+    ),
+    'entity_id' => array(
+      'entity_id',
+    ),
+    'revision_id' => array(
+      'revision_id',
+    ),
+    'language' => array(
+      'language',
+    ),
+  ),
+  'mysql_character_set' => 'utf8',
+));
+
+$connection->insert('field_revision_field_telephone')
+  ->fields(array(
+    'entity_type',
+    'bundle',
+    'deleted',
+    'entity_id',
+    'revision_id',
+    'language',
+    'delta',
+    'field_telephone_value',
+  ))
+  ->values(array(
+    'entity_type' => 'node',
+    'bundle' => 'test_content_type',
+    'deleted' => '0',
+    'entity_id' => '1',
+    'revision_id' => '1',
+    'language' => 'und',
+    'delta' => '0',
+    'field_telephone_value' => '99-99-99-99',
+  ))
+  ->execute();
 $connection->schema()->createTable('field_revision_field_term_entityreference', array(
   'fields' => array(
     'entity_type' => array(
@@ -45530,6 +45759,13 @@
   'module' => 'phone',
   'weight' => '0',
 ))
+->values(array(
+  'name' => 'MigrateTelephoneFieldHandler',
+  'type' => 'class',
+  'filename' => 'sites/all/modules/telephone/telephone.migrate.inc',
+  'module' => 'telephone',
+  'weight' => '0',
+))
 ->values(array(
   'name' => 'MigrateTranslationEntityHandler',
   'type' => 'class',
@@ -49415,6 +49651,10 @@
   'filename' => 'sites/all/modules/phone/tests/phone.za.test',
   'hash' => 'c5491ab663972aa23ae2f917a0fc605a6136f02e1b207d3fc650ed1f251359ee',
 ))
+->values(array(
+  'filename' => 'sites/all/modules/telephone/telephone.migrate.inc',
+  'hash' => '14561d51028bcf5b9e2f18654c3ba76eda444feb12ea6ac4e88fdbf4c1a35650',
+))
 ->values(array(
   'filename' => 'sites/all/modules/views/tests/comment/views_handler_argument_comment_user_uid.test',
   'hash' => 'b8b417ef0e05806a88bd7d5e2f7dcb41339fbf5b66f39311defc9fb65476d561',
@@ -53013,6 +53253,17 @@
   'weight' => '0',
   'info' => 'a:12:{s:4:"name";s:7:"Picture";s:11:"description";s:15:"Picture element";s:4:"core";s:3:"7.x";s:12:"dependencies";a:3:{i:0;s:6:"ctools";i:1;s:5:"image";i:2;s:11:"breakpoints";}s:5:"files";a:1:{i:0;s:27:"includes/PictureMapping.php";}s:9:"configure";s:26:"admin/config/media/picture";s:7:"package";s:7:"Picture";s:11:"stylesheets";a:1:{s:3:"all";a:1:{s:19:"picture_wysiwyg.css";s:45:"sites/all/modules/picture/picture_wysiwyg.css";}}s:5:"mtime";i:1544936288;s:7:"version";N;s:3:"php";s:5:"5.2.4";s:9:"bootstrap";i:0;}',
 ))
+->values(array(
+  'filename' => 'sites/all/modules/telephone/telephone.module',
+  'name' => 'telephone',
+  'type' => 'module',
+  'owner' => '',
+  'status' => '1',
+  'bootstrap' => '0',
+  'schema_version' => '0',
+  'weight' => '0',
+  'info' => 'a:12:{s:4:"name";s:9:"Telephone";s:11:"description";s:43:"Defines a field type for telephone numbers.";s:7:"package";s:6:"Fields";s:4:"core";s:3:"7.x";s:12:"dependencies";a:1:{i:0;s:5:"field";}s:5:"files";a:1:{i:0;s:21:"telephone.migrate.inc";}s:7:"version";s:14:"7.x-1.0-alpha1";s:7:"project";s:9:"telephone";s:9:"datestamp";s:10:"1389736105";s:5:"mtime";i:1389736105;s:3:"php";s:5:"5.2.4";s:9:"bootstrap";i:0;}',
+))
 ->values(array(
   'filename' => 'sites/all/modules/title/tests/title_test.module',
   'name' => 'title_test',
diff --git a/core/modules/migrate_drupal/tests/src/Kernel/d7/FieldDiscoveryTest.php b/core/modules/migrate_drupal/tests/src/Kernel/d7/FieldDiscoveryTest.php
index d3ac665d64..f952c1ff15 100644
--- a/core/modules/migrate_drupal/tests/src/Kernel/d7/FieldDiscoveryTest.php
+++ b/core/modules/migrate_drupal/tests/src/Kernel/d7/FieldDiscoveryTest.php
@@ -230,6 +230,9 @@ public function addAllFieldProcessesAltersData() {
                   'image_nodelink' => 'image',
                   'image_imagelink' => 'image',
                 ],
+                'telephone' => [
+                  'text_plain' => 'string',
+                ],
               ],
             ],
           ],
@@ -276,7 +279,7 @@ public function testGetAllFields() {
     $this->assertArrayHasKey('test_content_type', $actual_fields['node']);
     $this->assertSame(6, count($actual_fields['node']));
     $this->assertSame(6, count($actual_fields['comment']));
-    $this->assertSame(22, count($actual_fields['node']['test_content_type']));
+    $this->assertSame(23, count($actual_fields['node']['test_content_type']));
     foreach ($actual_fields as $entity_type_id => $bundles) {
       foreach ($bundles as $bundle => $fields) {
         foreach ($fields as $field_name => $field_info) {
diff --git a/core/modules/migrate_drupal_ui/tests/src/Functional/d7/MultilingualReviewPageTest.php b/core/modules/migrate_drupal_ui/tests/src/Functional/d7/MultilingualReviewPageTest.php
index b5fdc0239b..c06940c351 100644
--- a/core/modules/migrate_drupal_ui/tests/src/Functional/d7/MultilingualReviewPageTest.php
+++ b/core/modules/migrate_drupal_ui/tests/src/Functional/d7/MultilingualReviewPageTest.php
@@ -124,6 +124,7 @@ protected function getAvailablePaths() {
       'syslog',
       'system',
       'taxonomy',
+      'telephone',
       'term_depth',
       'text',
       'title',
diff --git a/core/modules/migrate_drupal_ui/tests/src/Functional/d7/Upgrade7Test.php b/core/modules/migrate_drupal_ui/tests/src/Functional/d7/Upgrade7Test.php
index f5178bc545..b31851678c 100644
--- a/core/modules/migrate_drupal_ui/tests/src/Functional/d7/Upgrade7Test.php
+++ b/core/modules/migrate_drupal_ui/tests/src/Functional/d7/Upgrade7Test.php
@@ -74,8 +74,8 @@ protected function getEntityCounts() {
       'contact_form' => 3,
       'contact_message' => 0,
       'editor' => 2,
-      'field_config' => 73,
-      'field_storage_config' => 55,
+      'field_config' => 74,
+      'field_storage_config' => 56,
       'file' => 3,
       'filter_format' => 7,
       'image_style' => 6,
@@ -99,7 +99,7 @@ protected function getEntityCounts() {
       'date_format' => 11,
       'entity_form_display' => 17,
       'entity_form_mode' => 1,
-      'entity_view_display' => 28,
+      'entity_view_display' => 29,
       'entity_view_mode' => 14,
       'base_field_override' => 4,
     ];
@@ -159,6 +159,7 @@ protected function getAvailablePaths() {
       'statistics',
       'system',
       'taxonomy',
+      'telephone',
       'text',
       'title',
       'user',
diff --git a/core/modules/telephone/migrations/state/telephone.migrate_drupal.yml b/core/modules/telephone/migrations/state/telephone.migrate_drupal.yml
index 0a39c8184a..d81274a849 100644
--- a/core/modules/telephone/migrations/state/telephone.migrate_drupal.yml
+++ b/core/modules/telephone/migrations/state/telephone.migrate_drupal.yml
@@ -1,3 +1,4 @@
 finished:
   7:
     phone: telephone
+    telephone: telephone
diff --git a/core/modules/telephone/src/Plugin/migrate/field/d7/TelephoneField.php b/core/modules/telephone/src/Plugin/migrate/field/d7/TelephoneField.php
new file mode 100644
index 0000000000..21abb171b7
--- /dev/null
+++ b/core/modules/telephone/src/Plugin/migrate/field/d7/TelephoneField.php
@@ -0,0 +1,35 @@
+<?php
+
+namespace Drupal\telephone\Plugin\migrate\field\d7;
+
+use Drupal\migrate_drupal\Plugin\migrate\field\FieldPluginBase;
+
+/**
+ * @MigrateField(
+ *   id = "telephone",
+ *   core = {7},
+ *   source_module = "telephone",
+ *   destination_module = "telephone"
+ * )
+ */
+class TelephoneField extends FieldPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFieldWidgetMap() {
+    // The widget IDs are identical in Drupal 7 and 8, so we do not need any
+    // mapping.
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFieldFormatterMap() {
+    return [
+      'text_plain' => 'string',
+    ];
+  }
+
+}
